### ISSUE TITLE
Optimizer state resets when calling fit method 

### ISSUE THREAD

**Issue by smehdia** [keras-team/keras#13750]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Ubuntu 18.04 LTS
- TensorFlow backend (yes / no):  Yes
- TensorFlow version:  1.13.2
- Keras version:  2.2.4
- Python version:  3.6.8
- CUDA/cuDNN version:  CUDA V10.0.130 / cuDNN 7
- GPU model and memory:  1080 Titan X (12 GB)

**Describe the current behavior**  
When the model is trained epoch by the epoch in a custom loop, it has different behavior in comparison with when the model is trained with just using 'fit' method. In fact, the optimizer state resets every time "fit" is called.
**Describe the expected behavior**  
The expected behavior is to have the same training loss through epochs for both methods.
**Code to reproduce the issue**  
Code 1: using a custom loop:

    batch_size = 128
    num_classes = 10
    epochs = 12
    
    # input image dimensions
    img_rows, img_cols = 28, 28
    
    # the data, split between train and test sets
    (x_train, y_train), (x_test, y_test) = mnist.load_data()
    
    if K.image_data_format() == 'channels_first':
        x_train = x_train.reshape(x_train.shape[0], 1, img_rows, img_cols)
        x_test = x_test.reshape(x_test.shape[0], 1, img_rows, img_cols)
        input_shape = (1, img_rows, img_cols)
    else:
        x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
        x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
        input_shape = (img_rows, img_cols, 1)
    
    x_train = x_train.astype('float32')
    x_test = x_test.astype('float32')
    x_train /= 255
    x_test /= 255
    print('x_train shape:', x_train.shape)
    print(x_train.shape[0], 'train samples')
    print(x_test.shape[0], 'test samples')
    
    # convert class vectors to binary class matrices
    y_train = keras.utils.to_categorical(y_train, num_classes)
    y_test = keras.utils.to_categorical(y_test, num_classes)
    
    opt = keras.optimizers.Adam(1e-3)
    
    model = Sequential()
    model.add(Conv2D(32, kernel_size=(3, 3),
                     activation='relu',
                     input_shape=input_shape))
    model.add(Conv2D(64, (3, 3), activation='relu'))
    model.add(MaxPooling2D(pool_size=(2, 2)))
    model.add(Dropout(0.25))
    model.add(Flatten())
    model.add(Dense(128, activation='relu'))
    model.add(Dropout(0.5))
    model.add(Dense(num_classes, activation='softmax'))
    
    model.load_weights('model.h5')
    
    for i in range(10):
    
        model.compile(loss=keras.losses.categorical_crossentropy,
                          optimizer=opt,
                          metrics=['accuracy'])
    
        model.fit(x_train, y_train,
                  batch_size=batch_size,
                  epochs=i+2,
                  verbose=1,
                  validation_data=(x_test, y_test), initial_epoch=i+1)

Code 2: using 'fit' for all epochs:
`
    model.fit(x_train, y_train,
              batch_size=batch_size,
              epochs=10,
              verbose=1,
              validation_data=(x_test, y_test))`
**Other info / logs**  
Logs for Custom Loop:
`Train on 60000 samples, validate on 10000 samples
	Epoch 2/2
	2020-02-05 13:14:32.774853: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library libcublas.so.10.0 locally
	60000/60000 [==============================] - 5s 78us/step - loss: 0.2442 - acc: 0.9248 - val_loss: 0.0564 - val_acc: 0.9819
	Train on 60000 samples, validate on 10000 samples
	Epoch 3/3
	60000/60000 [==============================] - 4s 64us/step - loss: 0.0988 - acc: 0.9707 - val_loss: 0.0378 - val_acc: 0.9881
	Train on 60000 samples, validate on 10000 samples
	Epoch 4/4
	60000/60000 [==============================] - 4s 64us/step - loss: 0.0767 - acc: 0.9767 - val_loss: 0.0304 - val_acc: 0.9905`
Code 2 outputs:
`Epoch 1/10
	2020-02-05 13:19:06.534620: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library libcublas.so.10.0 locally
	60000/60000 [==============================] - 5s 79us/step - loss: 0.2317 - acc: 0.9295 - val_loss: 0.0490 - val_acc: 0.9835
	Epoch 2/10
	60000/60000 [==============================] - 4s 59us/step - loss: 0.0823 - acc: 0.9758 - val_loss: 0.0364 - val_acc: 0.9869
	Epoch 3/10
	60000/60000 [==============================] - 4s 58us/step - loss: 0.0620 - acc: 0.9812 - val_loss: 0.0369 - val_acc: 0.9867`

**Comment by Saduf2019** [comment]
@smehdia 
We see that you are using old version of tensorflow which is not actively supported now, please verify on 2.x and open a new issue in case you face any errors, hence moving this to closed status.
